From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Mon, 31 Aug 2026 23:45:19 +0200
Subject: fix(busybox): do not install mount applet

The busybox `mount` applet does not resolve "auto" to the actual
filesystem and therefore passes `auto` directly to mount(2). If the
target partition's filesystem driver (e.g., `ext4`, `xfs`) is compiled
as a module and not yet loaded into `/proc/filesystems`, the kernel
skips `request_module()` and returns `EINVAL` (Invalid argument). For
example `test/test.sh debian:sid 10` fails with, because `ext4` is not
loaded:

```
systemd[1]: Mounting sysroot.mount - /sysroot...
mount: mounting /dev/disk/by-label/dracut on /sysroot failed: Invalid argument
```

Instead of adding a quirky workaround to probe the device and load the
relevant filesystem driver, do not install the `mount` applet.

Origin: upstream, https://github.com/dracut-ng/dracut/commit/9dcc127e2d190ae819110ce82f429a7d1d92c380
---
 modules.d/10busybox/module-setup.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules.d/10busybox/module-setup.sh b/modules.d/10busybox/module-setup.sh
index 9643c29..179d713 100755
--- a/modules.d/10busybox/module-setup.sh
+++ b/modules.d/10busybox/module-setup.sh
@@ -28,6 +28,11 @@ install() {
             # See https://github.com/vda-linux/busybox_mirror/issues/33
             continue
         fi
+        if [[ ${_path##*/} == mount ]]; then
+            # The busybox mount applet does not resolve "auto" to the actual filesystem.
+            # See https://github.com/vda-linux/busybox_mirror/issues/34
+            continue
+        fi
 
         # if busybox is built without CONFIG_FEATURE_INSTALLER=y, the list
         # has only plain names
